clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/BackgroundEventHandler.java

```diff
@@ -49,4 +49,9 @@ public void add(BackgroundEvent event) {
         log.trace("Enqueued event: {}", event);
         backgroundEventQueue.add(event);
     }
+
+    // Visible for testing
+    public Queue<BackgroundEvent> backgroundEventQueue() {
+        return backgroundEventQueue;
+    }
```

Comment:
I'm not terribly crazy about this approach as it can lead to circumvention of the logic. I understand it's just for tests, for which we do have the `ConsumerTestBuilder` approach which allows tests to get access to these kinds of objects directly.

Reply:
I had to take painkillers to stand the pain when I added that method because I totally agree with you but I did not find another way. I will reconsider.